go/common/logs.go

```diff
@@ -36,3 +36,12 @@ type LogsByRollupByID = map[rpc.ID]map[uint64][]*types.Log
 
 // EncLogsByRollupByID is identical to LogsByRollupByID, but with the logs encrypted as bytes.
 type EncLogsByRollupByID = map[rpc.ID]map[uint64][]byte
+
+// FilterCriteriaJSON is a structure that JSON-serialises to the expected format for log filter criteria.
+type FilterCriteriaJSON struct {
```

Comment:
Geth's `FilterCriteria` class has very annoying behaviour, whereby it implements its own JSON deserialiser that prevents it being round-tripped to JSON. Instead, we use this custom class that serialises correctly to be deserialised to a filter criteria later.